javav2/example_code/keyspaces/src/main/java/com/example/keyspace/ScenarioKeyspaces.java

```diff
@@ -0,0 +1,625 @@
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("14. Restore the table back to the previous state using the timestamp.");
+        System.out.println("Note that the restore operation can take up to 20 minutes.");
+        restoreTable(keyClient, keyspaceName, utc);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("15. Check for completion of the restore action.");
+        Thread.sleep(5000);
+        checkRestoredTable(keyClient, keyspaceName, "MovieRestore");
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("16. Delete both tables.");
+        deleteTable(keyClient, keyspaceName, tableName);
+        deleteTable(keyClient, keyspaceName, tableNameRestore);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("17. Confirm that both tables are deleted.");
+        checkTableDelete(keyClient, keyspaceName, tableName);
+        checkTableDelete(keyClient, keyspaceName, tableNameRestore);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("18. Delete the keyspace.");
+        deleteKeyspace(keyClient, keyspaceName);
+        System.out.println(DASHES);
+
+        System.out.println(DASHES);
+        System.out.println("The scenario has completed successfully.");
+        System.out.println(DASHES);
+    }
+
+    //snippet-start:[keyspace.java2.scenario.delete.keyspace.main]
+    public static void deleteKeyspace(KeyspacesClient keyClient, String keyspaceName) {
+        try {
+            DeleteKeyspaceRequest deleteKeyspaceRequest = DeleteKeyspaceRequest.builder()
+                .keyspaceName(keyspaceName)
+                .build();
+
+            keyClient.deleteKeyspace(deleteKeyspaceRequest);
+
+        } catch (KeyspacesException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+    //snippet-end:[keyspace.java2.scenario.delete.keyspace.main]
+
+    public static void checkTableDelete(KeyspacesClient keyClient, String keyspaceName, String tableName)throws InterruptedException {
+        try {
+            String status;
+            GetTableResponse response;
+            GetTableRequest tableRequest = GetTableRequest.builder()
+                .keyspaceName(keyspaceName)
+                .tableName(tableName)
+                .build();
+
+            // Keep looping until table cannot be found and exception is thrown.
+            while (true) {
+                response = keyClient.getTable(tableRequest);
+                status = response.statusAsString();
+                System.out.println(". The table status is "+status);
+                Thread.sleep(500);
+            }
+
+        } catch (KeyspacesException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());

```

Comment:
Should this specify a ResourceNotFoundException in order to return true? The epic specifies that exception, whereas the code here will catch on any exception.